pkg/enqueue/Symfony/DependencyInjection/TransportFactory.php

```diff
@@ -226,4 +222,16 @@ private function assertServiceExists(ContainerBuilder $container, string $servic
             throw new \InvalidArgumentException(sprintf('The service "%s" does not exist.', $serviceId));
         }
     }
+
+    private function fillLocator(ContainerBuilder $container, string $serviceName): void
+    {
+        $locatorId = 'enqueue.locator';
+
+        if ($container->hasDefinition($locatorId)) {
+            $locator = $container->getDefinition($locatorId);
+            $locator->replaceArgument(0, array_replace($locator->getArgument(0), [
```

Comment:
This could be a bit better: 

```php
<?php

$map = $locator->getArgument(0);
$map[$this->format($serviceName)] = $this->reference($serviceName));

$locator->replaceArgument(0, $map);
```